Letterbook.Adapter.ActivityPub/Signatures/InputCheckingVisitor.cs

```diff
@@ -0,0 +1,102 @@
+﻿
+using System.Diagnostics.CodeAnalysis;
+using NSign;
+using static NSign.Constants;
+using NSign.Signatures;
+using StructuredFieldValues;
+
+namespace Letterbook.Adapter.ActivityPub.Signatures;
+
+internal sealed class InputCheckingVisitor : ISignatureComponentVisitor
+{
+    private readonly HttpRequestMessage _context;
+    
+    public InputCheckingVisitor(HttpRequestMessage context)
+    {
+        _context = context;
+    }
+    
+    public bool Found { get; private set; } = true;
+
+    public void Visit(SignatureComponent component)
+    {
+        /* Intentionally left blank because signature components can't be unavailable */
+    }
+    
+    public void Visit(HttpHeaderComponent httpHeader)
+    {
+        Found &= TryGetHeaderValues(_context, httpHeader.ComponentName, out _);
```

Comment:
This part isn't timing sensitive. At least, I don't see how it could be. It's checking the message for the presence of the fields that should be signed over.